impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/serialization/SerializationTestCase.java

```diff
@@ -217,11 +217,12 @@ private void testCurrentFields(final Archive<?> payload, final Archive<?> roundt
      */
     private static SerializableView serializeAndDeserialize(final SerializableView archive) throws IOException, ClassNotFoundException {
         assert archive != null : "Archive must be specified";
+        //there are failing tests for JDK 8 with error message "ID not as expected after serialization" if we put this stream in try-with-resources
```

Comment:
```
final ByteArrayOutputStream byteOut = new ByteArrayOutputStream();
try (final ObjectOutputStream out = new ObjectOutputStream(byteOut)) {
    out.writeObject(archive);
}
try (ObjectInputStream in = new ObjectInputStream(new ByteArrayInputStream(byteOut.toByteArray()))) {
    return (SerializableView) in.readObject();
}
```

Reply:
Just a note here, but having the `ByteArrayOutputStream` inside the try clause is redundant. Documentation states that "Closing a ByteArrayOutputStream has no effect." and if you have a look inside the `ByteArrayOutputStream` you can see that the `close` method is empty.

But no harm in leaving it as is, just a fun fact